### ISSUE TITLE
Determining if on route

### ISSUE THREAD

**Issue by deathg0d** [tidwall/tile38#182]
I was trying to calculate if a vehicle is on a predefined route. For this I used several circular geofences on the route and checked if the course is as expected. Is there a better method to do this? Or would it be useful to somehow introduce linestring? So far I haven't seen any documentation regarding this, also it looks resource consuming and would require optimizations. Regardless, don't you think ability to check if a vehicle is in its route would be a good feature for tile38? Or would this be outside tile38's scope?

**Comment by Lars-Meijer** [comment]
I think what you are looking for is #79, there is however no functionality that currently supports this.

**Comment by deathg0d** [comment]
@Lars-Meijer Yes that is what I am looking for. For now I will use the polygon method. Thanks. However, it would be great to have feature as described in #79 as many open data have streets depicted as linestrings.

**Comment by Lars-Meijer** [comment]
> However, it would be great to have feature as described in #79 as many open data have streets depicted as linestrings.

I agree, it would be a useful feature.

I did some digging and found a similar feature in the Java Topology Suite (JTS), it's called buffer and its documentation can be found here: https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html#buffer(double)

Maybe this can be ported to go (or maybe it already has been ported), or maybe you can create a java program that converts linestrings to polygons with this.

**Comment by tidwall** [comment]
@deathg0d I'm sorry that the buffering feature is not a part of Tile38 yet. This will be an incredible addition. Both for geofencing and performing NEARBY/KNN queries on polygons. For now there's only workarounds.

Adding circles is one way, or ever a series of circles and/or polygons that simulate a buffer around the line. These geometries could then be added to one FeatureCollection that the geofence uses. There may be other ways too, and I know it's not ideal. :\

@Lars-Meijer Thanks for sharing the link. The JTS library looks very interesting. A potential target for porting to Go. 

Right now I'm knee-deep in getting the alpha branch for 2.0 together, but this feature is on my short list for "absolute next additions".

**Comment by deathg0d** [comment]
@Lars-Meijer Thanks a lot for the link. I will check it out too.
@tidwall No pressure :) We understand that you are trying to do a lot. Take your time. Your responsiveness to our queries already mean a lot to us.

**Comment by abhit011** [comment]
Isn't perpendicular distance is best approach to determine if its on route or not ? The length of the perpendicular is what determines your range ...
Abhi
!wq
